Kickstarter-iOS/Views/Controllers/DiscoveryNavigationHeaderViewController.swift

```diff
@@ -295,13 +312,64 @@ internal final class DiscoveryNavigationHeaderViewController: UIViewController {
     }
   }
 
+  @objc fileprivate func environmentSwitcherTapped() {
+    self.showEnvironmentActionSheet()
+  }
+
   @objc fileprivate func titleButtonTapped() {
     self.viewModel.inputs.titleButtonTapped()
   }
 
   @objc fileprivate func favoriteButtonTapped() {
     self.viewModel.inputs.favoriteButtonTapped()
   }
+
+  private func showEnvironmentActionSheet() {
+
+    let config = ServerConfig.environmentName(config: AppEnvironment.current.apiService.serverConfig)
```

Comment:
I wonder if we could simplify this a bit by changing up how we keep track of the environment types? If we added an enum:
```
public enum EnvironmentType: String {
  case production = "Production"
  case staging = "Staging"
  case local = "Local"
  
  static var allValues: [EnvironmentType] = [.production, .staging, .local]
}
```
and had it as a variable of the `ServerConfig` struct, then we could do:

```
EnvironmentType.allValues.forEach { environment in
      alert.addAction(
        UIAlertAction(title: environment.rawValue, style: .default) { [weak self] _ in
           // action handler
        }
      )
}
```
And we could access the current environment with: `AppEnvironment.current.apiService.serverConfig.environmentType.rawValue`. What do you think?

Reply:
I think this is cool and would avoid the need for the _unknown_ case [here](https://github.com/kickstarter/ios-oss/pull/370/files#diff-2cab5a0cd36c9da79573767a85181bf1R83), Also, Swift 4.2 has `allCases` [baked in](https://github.com/apple/swift-evolution/blob/master/proposals/0194-derived-collection-of-enum-cases.md) so we would be able to drop the array later on, in fact perhaps we want to name that `allCases` in anticipation of just using the built-in static var later on.

Reply:
I'd added a property called `environmentName` previously that following a similar idea, but for having it as a property, was causing a bug on which the environment wasn't being shown before we changed the environment for the first time (what is not a big deal, since this is an internal build). One solution for this could be to re-add a property and set its default value to `.production`.